packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -388,22 +395,21 @@ contract Ethernauts is ERC721Enumerable, Ownable {
             _generateRandomNumber();
         }
 
-        _mint(to, tokenId);
+        _safeMint(to, tokenId);
     }
 
-    function _mint(address to, uint256 tokenId) internal virtual override {
+    function _safeMint(address to, uint256 tokenId) internal virtual override {
         if (totalSupply() >= maxTokens) {
             revert TotalSupplyError({total: totalSupply(), max: maxTokens});
         }
-        super._mint(to, tokenId);
+
+        super._safeMint(to, tokenId);
```

Comment:
by calling safeMint we actually open the possibilities to reentrancy as safeMint does 
`require(_checkOnERC721Received(address(0), to, tokenId, _data), "ERC721: transfer to non ERC721Receiver implementer");`

Reply:
Yep, I am aware of this. But do you think we're still vulnerable to re-entrancy now that the reentrancy guard has been implemented?

Reply:
Not necessarily! I think we have reentrancy guard + checks effects interactions, we should be good!